### ISSUE TITLE
Optional use of 16 bit words? 

### ISSUE THREAD

**Issue by seaglade** [BenWiederhake/splaycompress#1]
Some testing would be needed to see if it provides a tangible benefit rather than just a theoretical one, but I wonder if there could be a format (maybe `jan16`?) that *does* use 16 bit words. The README mentions deciding against this due to the memory requirements, but perhaps it could be enabled as a CLI flag and detected automatically via the filemagic and/or extension, allowing users who know their system can handle it to trade off compression ratio for memory usage. I could maybe take a crack at implementing this at some point.

**Comment by BenWiederhake** [comment]
Hi there, thanks for your interest!

The actual problem I ran into isn't the actual memory requirement itself. Afterall, it's only 65535 internal nodes with two arms that store 2+1 bytes each, which should get padded to 4 bytes per arm, so that's … half a megabyte. That should easily fit on most systems.

However, the way I did it, and the way it currently works, is to store this *on the stack*, and *then* half a megabyte suddenly is a very large chunk of memory. That's why I saw a stack exhaustion error when trying to run it this way.

You're very welcome to implement it! In an actual tool the format could be auto-detected, yes. For now, I'd like to keep the format absolutely bare-bones, for the usecase of extremely-small blobs of data, below 500 bytes.
